sklearn/cross_decomposition/pls_.py

```diff
@@ -406,11 +416,11 @@ def predict(self, X, copy=True):
         be an issue in high dimensional space.
         """
         check_is_fitted(self, 'x_mean_')
+        X = check_array(X, copy=self.copy)
         # Normalize
         if copy:
-            Xc = (np.asarray(X) - self.x_mean_)
+            Xc = X - self.x_mean_
         else:
-            X = np.asarray(X)
             Xc -= self.x_mean_
             Xc /= self.x_std_
```

Comment:
It seems that there is a spurious indent on this line: the data is not scaled by `self.x_std_` when `copy=False` (agreed the bug was already there prior to you PR).